internal/pkg/deploy/cloudformation/stack/scheduled_job_test.go

```diff
@@ -0,0 +1,330 @@
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+			conf := &ScheduledJob{
+				wkld: &wkld{
+					name: aws.StringValue(testScheduledJobManifest.Name),
+					env:  testJobEnvName,
+					app:  testJobAppName,
+					rc: RuntimeConfig{
+						ImageRepoURL: testJobImageRepoURL,
+						ImageTag:     testJobImageTag,
+					},
+				},
+				manifest: testScheduledJobManifest,
+			}
+			tc.mockDependencies(t, ctrl, conf)
+
+			// WHEN
+			template, err := conf.Template()
+
+			// THEN
+			if tc.wantedError != nil {
+				require.EqualError(t, err, tc.wantedError.Error())
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, tc.wantedTemplate, template)
+			}
+		})
+	}
+}
+
+func TestScheduledJob_awsSchedule(t *testing.T) {
+	testCases := map[string]struct {
+		inputSchedule  string
+		wantedSchedule string
+		wantedError    error
+	}{
+		"simple rate": {
+			inputSchedule:  "@every 1h30m",
+			wantedSchedule: "rate(90 minutes)",
+		},
+		"missing schedule": {
+			inputSchedule: "",
+			wantedError:   errors.New("missing required field schedule in manifest for job mailer"),
+		},
+		"one minute rate": {
+			inputSchedule:  "@every 1m",
+			wantedSchedule: "rate(1 minute)",
+		},
+		"truncate to minute if using small units": {
+			inputSchedule:  "@every 61000ms",
+			wantedSchedule: "rate(1 minute)",
+		},
+		"malformed rate": {
+			inputSchedule: "@every 402 seconds",
+			wantedError:   errors.New("schedule @every 402 seconds for job mailer is not a valid cron expression or definition string"),
+		},
+		"malformed cron": {
+			inputSchedule: "every 4m",
+			wantedError:   errors.New("schedule every 4m for job mailer is not a valid cron expression or definition string"),
+		},
+		"correctly converts predefined schedule": {
+			inputSchedule:  "@daily",
+			wantedSchedule: "cron(0 0 * * ? *)",
+		},
+		"unrecognized predefined schedule": {
+			inputSchedule: "@minutely",
+			wantedError:   errors.New("schedule @minutely for job mailer is not a valid cron expression or definition string"),
+		},
+		"correctly converts cron with all asterisks": {
+			inputSchedule:  "* * * * *",
+			wantedSchedule: "cron(* * * * ? *)",
+		},
+		"correctly converts cron with specified DOW": {
+			inputSchedule:  "* * * * MON-FRI",
+			wantedSchedule: "cron(* * ? * MON-FRI *)",
+		},
+		"correctly converts crom with specified DOM": {
+			inputSchedule:  "* * 1 * *",
+			wantedSchedule: "cron(* * 1 * ? *)",
+		},
+		"correctly increments 0-indexed DOW": {
+			inputSchedule:  "* * ? * 2-6",
+			wantedSchedule: "cron(* * ? * 3-7 *)",
+		},
+		"returns error if both DOM and DOW specified": {
+			inputSchedule: "* * 1 * SUN",
+			wantedError:   errors.New("parse cron schedule: cannot specify both DOW and DOM in cron expression"),
+		},
+		"returns error if fixed interval less than one minute": {
+			inputSchedule: "@every 50s",
+			wantedError:   errors.New("parse fixed interval: duration must be >= 1 minute"),
+		},
+		"truncates to minute (round down)": {
```

Comment:
So much impressive work in this PR! 
Small question here: should we tell the user that their schedule is being rounded down? (Maybe it's already in here and I'm not seeing it.)

Reply:
Hmm. I think the default behavior in all cases should be to round to the nearest minute, rather than throw an error, so perhaps if we change the behavior in `job init` to match we can alter our messaging to make it clear.